### DISCUSSION TITLE
Data collection randomly stalling – where are the logs located?

### DISCUSSION THREAD

**Discussion by Green-Pool** [jasonacox/Powerwall-Dashboard#498] [Q&A] [answered]
I have been running Powerwall-Dashboard for about 18 months.  It is an awesome tool to understand what is happening with the Powerwall.

Generally, it has been working ok.  Of late, it had stopped collecting data from the Powerwall.  No obvious pattern in the stop time.  I can still access the dashboard, but the back end has stopped collecting data.  The easiest way to get it running again is the re-run the upgrade.sh script.  It seems to work fine for another week or so, then it stalls again.

I am running the Powerwall-Dashboard on a Linux Mint VM.  The VM also has a LAMP install that is collecting data from the solar PV inverters and some local weather data.  

Does the Powerwall-Dashboard generate logs that I can go digging through?

**Comment by jasonacox** [comment, marked as answer]
> Does the Powerwall-Dashboard generate logs that I can go digging through?

Hi @Green-Pool - yes, it does, each component will have different logs, but I suspect the issue should show up here:

```
docker logs pypowerwall
```

**Reply by Green-Pool** [reply]
Awesome.  Thanks for the pointer.  I just had a quick look, and presently there is nothing there of note.  I suspect that the reinstall deleted the prior logs.  Next time the process barfs, I will have a look at this log location.

**Reply by jasonacox** [reply]
Yes you right. The logs will be removed when the container is removed (which is what a reinstall would do).
